client-course-schedulizer/src/components/Tabs/FacultyLoads/FacultyLoads.tsx

```diff
@@ -0,0 +1,173 @@
+      });
+    });
+  });
+  return newTableData.map((row) => {
+    return {
+      ...row,
+      totalHours: (row.fallHours || 0) + (row.springHours || 0) + (row.summerHours || 0),
+    };
+  });
+};
+
+export const FacultyLoads = () => {
+  const { schedule } = useContext(ScheduleContext);
+
+  const data = useMemo<TableData[]>(() => {
+    return createTable(schedule);
+  }, [schedule]);
+
+  // TODO: Add Other Duties/Hours and Load Notes
+  const columns = useMemo<Column<TableData>[]>(() => {
+    return [
+      { Header: "Faculty", accessor: "faculty" },
+      { Header: "Fall Course Sections", accessor: "fallCourseSections" },
+      { Header: "Fall Hours", accessor: "fallHours" },
+      { Header: "Spring Course Sections", accessor: "springCourseSections" },
+      { Header: "Spring Hours", accessor: "springHours" },
+      { Header: "Summer Course Sections", accessor: "summerCourseSections" },
+      { Header: "Summer Hours", accessor: "summerHours" },
+      { Header: "Total Hours", accessor: "totalHours" },
+    ];
+  }, []);
+  const tableInstance = useTable({ columns, data });
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = tableInstance;
+
+  // https://react-table.tanstack.com/docs/quick-start
+  return (
+    // apply the table props
+    <TableContainer component={Paper}>
+      <Table {...getTableProps()}>
+        <TableHead>
+          {
+            // Loop over the header rows
+            headerGroups.map((headerGroup) => {
+              return (
+                // Apply the header row props
+                <TableRow {...headerGroup.getHeaderGroupProps()} key={headerGroup.id}>
+                  {
+                    // Loop over the headers in each row
+                    headerGroup.headers.map((column) => {
+                      return (
+                        // Apply the header cell props
+                        <TableCell {...column.getHeaderProps()} key={column.id}>
+                          {
+                            // Render the header
+                            column.render("Header")
+                          }
+                        </TableCell>
+                      );
+                    })
+                  }
+                </TableRow>
+              );
+            })
+          }
+        </TableHead>
+        {/* Apply the table body props */}
+        <TableBody {...getTableBodyProps()}>
+          {
+            // Loop over the table rows
+            rows.map((row) => {
+              // Prepare the row for display
+              prepareRow(row);
+              return (
+                // Apply the row props
+                <TableRow {...row.getRowProps()} key={row.id}>
+                  {
+                    // Loop over the rows cells
+                    row.cells.map((cell) => {
+                      // Apply the cell props
+                      return (
+                        <TableCell {...cell.getCellProps()} key={cell.value}>
```

Comment:
Not sure which key is giving duplicates, but this one potentially will give duplicates. You'll have to find a unique value for the key, perhaps stringifying the JSON object like `JSON.stringify(cell)` would guarantee uniqueness. Unique keys are important for optimized updating to the DOM.

Reply:
I'm realizing that this issue is just eslint being dumb. `getCellProps()` and the other getter methods to get the props returns a unique key prop, so it shouldn't be necessary to specify a key. Is it alright to just disable that eslint error for this file?

Reply:
@Jonri2, yup, it's fine to disable that key rule for the file. I still don't know what's best for ESLint. Also, you could call `const cellProps = cell.getCellProps()` and then do `<TabelCell key={cellProps.key} {...cellProps}>` if you prefer this way.